File: Backend/dockerFiles/solutionC/DockerFile.cpp
FROM gcc:latest

RUN apt-get update && apt-get install -y \
    pkg-config \
    libssl-dev \
    libcriterion-dev \
    build-essential \
    libpthread-stubs0-dev \
    && rm -rf /var/lib/apt/lists/*

WORKDIR /usr/local/solution

COPY solutionC/solution.c /usr/local/solution/solution.c
COPY solutionC/tests.c /usr/local/solution/tests.c

CMD ["bash"]

File: Backend/dockerFiles/solution/DockerFile.java
# Use OpenJDK 11 with Maven installed
FROM maven:3.8.4-openjdk-11-slim

# Install necessary dependencies
RUN apt-get update

# Copy project source code and Maven configuration
COPY solution/src /usr/local/solution/src
COPY solution/pom.xml /usr/local/solution/pom.xml

# Set the working directory inside the container
WORKDIR /usr/local/solution

# Install project dependencies
RUN mvn clean install

# Run tests using Maven
RUN mvn clean install test

# Set the entrypoint to bash so you can run commands interactively
CMD ["bash"]
